### ISSUE TITLE
Move the snake

### ISSUE THREAD

**Issue by nightwolf93** [circa94/JSlither#2]
Next step the movement ? :)
I have found the G packet, but i don't know how to use correctly do u have any idea ?

**Comment by eralpcicekdagi** [comment]
hello how can i learn my port

**Comment by circa94** [comment]
@eralpcicekdagi default port is 8080

@nightwolf93 i think i will have the movement done tomorrow :) It should be make with packet "e"

**Comment by nightwolf93** [comment]
The "e" ? Is for the the direction right ? With the "g" i can move my snake but it's slow xD

**Comment by eralpcicekdagi** [comment]
is new update work?

**Comment by circa94** [comment]
No. But now data is sendig correctly. We need to do a lot of more math...

**Comment by circa94** [comment]
anyone find out smth new?

**Comment by GrumpyOet** [comment]
i think i know why you cannot move you are actually always out the circle of the map

**Comment by iioel** [comment]
@GrumpyOet no, you are into the circle, the mini-map is just buggy. You can install [this](http://slitherio.org/slitio-mod-v2-0.html) script and you can see your X and Y axis.

**Comment by circa94** [comment]
this is also a good point. we should find out, how we can display the snake correctly on the minimap

**Comment by nightwolf93** [comment]
var incX = ((float)Math.Cos((Math.PI / 180) \* this.MySnake.CurrentAngle));
var incY = ((float)Math.Sin((Math.PI / 180) \* this.MySnake.CurrentAngle));

That for calculate the good increase vector for snake direction
(You need to convert you angle to radian like this)

**Comment by Kogs** [comment]
I just found this out at the moment. Now I read your comment ^^ Could save me hours if a had read this earlier. Damn

**Comment by nightwolf93** [comment]
Read your github notification haaha :+1: 
Btw you have found how to set the direction of the snake ?

**Comment by Kogs** [comment]
not really. I had the Problems with the movement (wich is working now but not perfect at all). The next Thing i want to try is the minimap. Because theres something wrong too

**Comment by nightwolf93** [comment]
My minimap work i think
![image](https://cloud.githubusercontent.com/assets/2054618/14653651/3cb0ff6e-067a-11e6-92c7-936353799b0a.png)

**Comment by Kogs** [comment]
IDK.
oh your movements looks good :D I only see a moving head in our server

**Comment by nightwolf93** [comment]
I think we have the same, just i have put some part in my snake haha

**Comment by Kogs** [comment]
Yes i added some parts and now i see the tail

**Comment by nightwolf93** [comment]
Also when u send the update position of the snake, the position is incorrect for u too ? is not the same of the new snake packet ?

**Comment by Kogs** [comment]
We have some problems with the position. On 2 clients the position of the snakes are not the same. Maybe only visual idk. Also the Rotation of the snake is missing. So the snake is head banging the hole time and the tail follows not correctly.

**Comment by nightwolf93** [comment]
Okay we need some more informations for understand how its work

**Comment by Kogs** [comment]
@nightwolf93 did you found something out?

**Comment by nightwolf93** [comment]
Yeah, how to set the speed of the snake correctly. But nothing more i didn't have the time :/

**Comment by RowanHarley** [comment]
The movement is still very buggy and it says that my snake spawned outside the map. It also says my score on the leaderboards 23k but the score in the bottom left is 150

**Comment by Kogs** [comment]
we know that. The readme of this repo also says that we are still in development.
